web/src/list_widget.ts

```diff
@@ -468,9 +474,11 @@ export function create<Key = unknown, Item = Key>(
                 widget.clean_redraw();
                 return;
             }
-            if (!opts.filter.predicate(item)) {
+
+            if (!opts.filter?.predicate?.(item, meta.filter_value)) {
```

Comment:
I think we want `!.` or assertions here. `?.` makes it silently return when either `filter` or `predicate` is missing. It's better to fail here to indicate that there is a programming error.